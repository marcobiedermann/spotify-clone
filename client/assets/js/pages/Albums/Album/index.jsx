import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Helmet } from 'react-helmet';
import { connect } from 'react-redux';
import {
  Route,
  Switch,
} from 'react-router-dom';
import { bindActionCreators } from 'redux';
import TracksPage from './Tracks';
import { fetchAlbum } from '../../../actions/albums';
import Album from '../../../components/Album';
import Error from '../../../components/Error';
import Loader from '../../../components/Loader';

export class AlbumPage extends Component {
  componentDidMount() {
    const {
      accessToken,
      fetchAlbum,
      match,
    } = this.props;

    fetchAlbum(accessToken, match.params.album_id);
  }

  render() {
    const {
      album,
      error,
      isLoading,
      match,
    } = this.props;

    return (
      <Switch>
        <Route
          path={`${match.url}/tracks`}
          component={TracksPage}
        />
        <Route
          path={match.url}
          component={() => {
            if (error) {
              return <Error>{error.message}</Error>;
            }

            if (isLoading) {
              return <Loader />;
            }

            return (
              <div className="album-page">
                <Helmet>
                  <title>{album.name}</title>
                </Helmet>
                <Album {...album} />
              </div>
            );
          }}
        />
      </Switch>
    );
  }
}

AlbumPage.propTypes = {
  accessToken: PropTypes.string,
  album: PropTypes.shape({
    name: PropTypes.string,
  }),
  error: PropTypes.shape({
    message: PropTypes.string,
  }),
  fetchAlbum: PropTypes.func,
  isLoading: PropTypes.bool,
  match: PropTypes.shape({
    params: PropTypes.shape({
      album_id: PropTypes.string,
    }),
    url: PropTypes.string,
  }),
};

AlbumPage.defaultProps = {
  accessToken: '',
  album: {
    name: '',
  },
  error: {
    message: '',
  },
  fetchAlbum: () => {},
  isLoading: false,
  match: {
    params: {
      album_id: '',
    },
    url: '',
  },
};

const mapStateToProps = state => ({
  ...state,
  album: state.albums.album,
  error: state.albums.error,
  isLoading: state.albums.isLoading,
});

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    fetchAlbum,
  },
  dispatch,
);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AlbumPage);
