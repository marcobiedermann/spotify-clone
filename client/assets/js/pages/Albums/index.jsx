import PropTypes from 'prop-types';
import React from 'react';
import {
  Route,
  Switch,
} from 'react-router-dom';
import AlbumPage from './Album';

const AlbumsPage = ({ match }) => (
  <Switch>
    <Route
      path={`${match.url}/:album_id`}
      component={AlbumPage}
    />
    <Route
      path={`${match.url}`}
      component={() => (
        <div>
          Albums
        </div>
      )}
    />
  </Switch>
);

AlbumsPage.propTypes = {
  match: PropTypes.shape({
    url: PropTypes.string,
  }),
};

AlbumsPage.defaultProps = {
  match: {
    url: '',
  },
};

export default AlbumsPage;
