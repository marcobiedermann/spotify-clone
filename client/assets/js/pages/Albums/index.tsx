import React, { FC } from 'react';
import { Helmet } from 'react-helmet';
import { RouteChildrenProps } from 'react-router-dom';

const AlbumsPage: FC<RouteChildrenProps> = (props) => {
  return (
    <>
      <Helmet>
        <title>Albums</title>
      </Helmet>
      Albums
    </>
  );
};

export default AlbumsPage;
