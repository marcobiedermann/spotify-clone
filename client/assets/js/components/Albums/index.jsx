import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import Album from '../Album';
import styles from './style.css';

class Albums extends PureComponent {
  static propTypes = {
    albums: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
      }),
    ),
  };

  static defaultProps = {
    albums: [],
  };

  render() {
    const { albums } = this.props;

    return (
      <ul className={styles.albums}>
        {albums.map(album => (
          <li key={album.id}>
            <Album {...album} />
          </li>
        ))}
      </ul>
    );
  }
}

export default Albums;
