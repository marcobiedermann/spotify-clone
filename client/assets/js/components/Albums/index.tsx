import classNames from 'classnames';
import React, { FC } from 'react';
import Album, { AlbumProps } from '../Album';
import styles from './style.css';

export interface AlbumsProps {
  className?: string;
  items: AlbumProps[];
}

const Albums: FC<AlbumsProps> = (props) => {
  const { className, items } = props;

  return (
    <ul className={classNames(className, styles.albums)}>
      {items.map((album) => (
        <li key={album.id}>
          <Album {...album} />
        </li>
      ))}
    </ul>
  );
};

export default Albums;
