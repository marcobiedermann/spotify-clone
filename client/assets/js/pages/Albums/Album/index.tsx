import React, { FC } from 'react';
import { Helmet } from 'react-helmet';
import { RouteChildrenProps, useParams } from 'react-router-dom';
import useSWR from 'swr';
import Album from '../../../components/Album';
import Error from '../../../components/Error';
import Loader from '../../../components/Loader';

interface Params {
  albumId: string;
}

const AlbumPage: FC<RouteChildrenProps> = () => {
  const { albumId } = useParams<Params>();
  const { data, error } = useSWR(`/v1/albums/${albumId}`);

  if (error) {
    return <Error>{error.message}</Error>;
  }

  if (!data) {
    return <Loader />;
  }

  const { name } = data;

  return (
    <>
      <Helmet>
        <title>{name}</title>
      </Helmet>
      <Album {...data} />
    </>
  );
};

export default AlbumPage;
