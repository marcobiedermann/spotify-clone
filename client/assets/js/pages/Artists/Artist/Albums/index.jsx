import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Albums from '../../../../components/Albums';

class AlbumsPage extends Component {
  componentDidMount() {
    const { accessToken, fetchArtistAlbums } = this.props;

    fetchArtistAlbums(accessToken, '20JZFwl6HVl6yg8a4H3ZqK');
  }

  render() {
    const { albums } = this.props;

    return (
      <Albums albums={albums} />
    );
  }
}

AlbumsPage.propTypes = {
  accessToken: PropTypes.string,
  albums: PropTypes.arrayOf(),
  fetchArtistAlbums: PropTypes.func,
};

AlbumsPage.defaultProps = {
  accessToken: '',
  albums: [],
  fetchArtistAlbums: () => {},
};

export default AlbumsPage;
