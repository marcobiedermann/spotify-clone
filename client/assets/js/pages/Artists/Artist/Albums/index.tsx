import React, { Component } from 'react';
import { Helmet } from 'react-helmet';
import { connect } from 'react-redux';
import { RouteChildrenProps } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import { fetchArtistAlbums } from '../../../../actions/artists';
import Albums, { AlbumsProps } from '../../../../components/Albums';
import Error from '../../../../components/Error';
import Loader from '../../../../components/Loader';

export interface AlbumsPageProps extends RouteChildrenProps {
  accessToken: string;
  albums: AlbumsProps;
  error: {
    message: string;
  };
  fetchArtistAlbums: any;
  isLoading: boolean;
}

export class AlbumsPage extends Component<AlbumsPageProps> {
  componentDidMount() {
    const { accessToken, fetchArtistAlbums } = this.props;

    fetchArtistAlbums(accessToken, '20JZFwl6HVl6yg8a4H3ZqK');
  }

  render() {
    const { albums, error, isLoading } = this.props;

    if (error) {
      return <Error>{error.message}</Error>;
    }

    if (isLoading) {
      return <Loader />;
    }

    return (
      <>
        <Helmet>
          <title>Albums</title>
        </Helmet>
        <Albums {...albums} />
      </>
    );
  }
}

const mapStateToProps = (state) => ({
  ...state,
  albums: state.artists.albums,
  error: state.artists.error,
  isLoading: state.artists.isLoading,
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchArtistAlbums,
    },
    dispatch,
  );

export default connect(mapStateToProps, mapDispatchToProps)(AlbumsPage);
