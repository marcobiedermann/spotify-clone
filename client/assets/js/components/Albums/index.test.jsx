import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Albums from '.';

describe('Albums component', () => {
  const albums = [
    {
      id: '123456',
    },
  ];

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter><Albums albums={albums} /></MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
