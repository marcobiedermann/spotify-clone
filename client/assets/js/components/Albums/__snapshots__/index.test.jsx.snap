// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums component renders correctly 1`] = `
<ul
  className="albums"
>
  <li>
    <div>
      <figure>
        <figcaption>
          <h3>
            <a
              href="/albums/123456"
              onClick={[Function]}
            >
              
            </a>
          </h3>
        </figcaption>
      </figure>
      <table>
        <tbody />
      </table>
    </div>
  </li>
</ul>
`;
